python/ray/tune/trial_runner.py

```diff
@@ -129,6 +128,7 @@ def step(self):
 
             if self.is_finished():
                 self._server.shutdown()
+        self.trial_executor.on_step_end()
```

Comment:
what is this for?

Reply:
it's a hook called every time the function step() finished. 
Maybe for other framework to report something at the end of step(), just like ray need to update the available resources at the beginning of step().